src/common/integration_test_base_class.py

```diff
@@ -0,0 +1,79 @@
+# standard library
+import unittest
+
+# third party
+import mysql.connector
+
+# first party
+from delphi.epidata.client.delphi_epidata import Epidata
+from delphi.epidata.server._limiter import limiter
+
+
+class BasicIntegrationTest(unittest.TestCase):
+    """Basic integration test class"""
+
+    def __init__(self, methodName: str = "runTest") -> None:
+        super().__init__(methodName)
+        self.delete_from_tables_list = []
+        self.truncate_tables_list = []
+        self.create_tables_list = []
+        self.role_name = None
+        self.epidata = Epidata
+        self.epidata.BASE_URL = "http://delphi_web_epidata/epidata/api.php"
+        self.epidata.auth = ("epidata", "key")
+
+    def delete_from_table(self, cur, table_name: str) -> None:
+        cur.execute(f"DELETE FROM `{table_name}`")
+
+    def truncate_table(self, cur, table_name: str) -> None:
+        cur.execute(f"TRUNCATE TABLE `{table_name}`")
+
+    def create_table(self, cur, create_table_stmt: str):
+        cur.execute(create_table_stmt)
+
+    def create_key_with_role(self, cur, role_name: str):
+        cur.execute("TRUNCATE TABLE `user_role`")
+        cur.execute("TRUNCATE TABLE `user_role_link`")
+        cur.execute(f'INSERT INTO `api_user`(`api_key`, `email`) VALUES("{role_name}_key", "{role_name}_email")')
+        cur.execute(f'INSERT INTO `user_role`(`name`) VALUES("{role_name}")')
+        cur.execute(
+            f'INSERT INTO `user_role_link`(`user_id`, `role_id`) SELECT `api_user`.`id`, 1 FROM `api_user` WHERE `api_key`="{role_name}_key"'
```

Comment:
```suggestion
            f'INSERT INTO `user_role_link`(`user_id`, `role_id`) SELECT `api_user`.`id`, `user_role`.`id` FROM `api_user` JOIN `user_role` WHERE `api_user`.`api_key`="{role_name}_key" AND `user_role`.`name`="{role_name}"'
```
without the hardcoded role id of "1", this should be more flexible/generalizable